lib/perl/Genome/Model/Command/Admin/PurgeSoftwareResultsFromAnalysisProject.pm

```diff
@@ -0,0 +1,171 @@
+        pattern   => UR::Context->date_template(),
+    );
+
+    my $now_str = UR::Context->now();
+    my $updated_at_str = $anp->updated_at;
+
+    my $dt1 = $strp->parse_datetime($updated_at_str);
+    my $dt2 = $strp->parse_datetime($now_str);
+
+    my $anp_updated_duration = $dt1->delta_days($dt2);
+
+    my $duration_to_retain = DateTime::Duration->new(
+        days        => $self->days_to_retain,
+    );
+
+    return ( DateTime::Duration->compare( $anp_updated_duration, $duration_to_retain ) == -1 );
+}
+
+
+sub _get_lock_for_analysis_project {
+    my($self, $anp) = @_;
+
+    my $unlocker;
+    if ($ENV{UR_DBI_NO_COMMIT}) {
+        $unlocker = sub {};
+
+    } else {
+        my $lock = Genome::Sys::LockProxy->new(
+            resource => 'genome model admin purge-software-results-from-analysis-project' . $anp->id,
+            scope => 'site',
+        )->lock(max_try => 1);
+
+        die $self->error_message('Cannot get lock for analysis project '.$anp->id) unless $lock;
+
+        $unlocker = sub {
+            $lock->unlock();
+        };
+    }
+    return $unlocker;
+}
+
+sub purge_one_analysis_project {
+    my($self, $anp) = @_;
+
+    if ($anp->is_cle) {
+        die $self->error_message('Failed to process CLE analysis project: '. $anp->id);
+    }
+
+    if ( $self->analysis_project_is_old_enough_to_purge($anp) ) {
+        $self->warning_message('Analysis project \''. $anp->id .'\' was updated at '. $anp->updated_at .' which is less than the '. $self->days_to_retain .' days to retain disabled results');
+        return;
+    }
+
+    my $dbh = Genome::DataSource::GMSchema->get_default_handle();
+    die $self->error_message('Cannot get default database handle') unless $dbh;
+
+    my $unlocker = $self->_get_lock_for_analysis_project($anp);
+
+    my $total_kb_purged = 0;
+    my $software_result_count = 0;
+
+    my $unlock_and_print_report = Scope::Guard->new(sub {
+        $unlocker->();
+        $self->status_message('Removed %d GB from %d software results',
+                                int($total_kb_purged / KB_IN_ONE_GB),
+                                $software_result_count);
+    });
+
+    my $sth = $dbh->prepare($sql);
+    die $self->error_message("preparing SQL failed : $DBI::errstr") unless $sth;
+
+    $sth->execute($anp->id);
+
+    while (my $data = $sth->fetchrow_hashref()) {
+        my $sr = Genome::SoftwareResult->get($data->{result_id});
+        die $self->error_message('Cannot get software result '. $data->{result_id}) unless $sr;
+
+        my $reason = 'Expunge software result uniquely used by model from disabled config item ('. $data->{profile_item_id} .') for analysis project \''. $data->{anp_name} .'\' ('. $data->{anp_id} .')';
+
+        $software_result_count++;
+        $total_kb_purged += $data->{kilobytes_requested};
+
+        if ($self->dry_run) {
+            $self->warning_message('Dry run, not removing software result '.$sr->id);
```

Comment:
What about having the same message with prepended with 'Would remove' and Removing', like 'git clean'? That would you can see what would happen on a 'wet_run' :smile_cat: